src/platform/src/main/java/org/geoserver/platform/ModuleStatusImpl.java

```diff
@@ -82,6 +84,14 @@ public ModuleStatusImpl(String module, String name) {
         this.isEnabled = true;
     }
 
+    public ModuleStatusImpl(String module, String name, Capability capability) {
```

Comment:
I guess you used a constructor to pass down the capability in spring right? That say I would not follow this approach. The GeoFence module implementation of  ``ModuleStatus`` should hard-coded the supported capabilities.

Reply:
fixed